### PR TITLE
added the first endpoint

### PR DESCRIPTION
**Pull request by richlloydmiles**
## Result

- Added the first endpoint (this is still WIP)

### CODE REVIEW

File: api/post/single-post.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    $result['breadcrumbs'][0]['name'] = 'Home';
+    $result['breadcrumbs'][0]['slug'] = '/';
+    $result['breadcrumbs'][1]['name'] = 'Blog';
+    $result['breadcrumbs'][1]['slug'] = '/travel-blog';
+    $result['breadcrumbs'][2]['name'] = 'Africa\'s best family road trips: Namibia';
+    $result['breadcrumbs'][2]['slug'] = '/africas-best-family-road-trips-namibia';
+    $result['categories'] = 'Travel Advice';
+
+    //sets the cached result
+    wp_cache_set('api_single_post', $result);
```
Comment (RollYourOwnEd) [inline api/post/single-post.php]: Isn't this cache-ing too generally? (not saying it is... asking) to me I would assume we need to cache that specific post id having been requested?

File: api/post/single-post.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+add_action('rest_api_init', function () {
+  register_rest_route('api/', '/post/(?P<post_id>\d+)', [
+    'methods' => 'GET',
+    'callback' => 'api_get_single_post'
+    ]);
+});
+
+/**
+ * Gets a route for a single post api endpoint
+ * @param  Array $data all the GET data that is send with API request
```
Comment (RollYourOwnEd) [inline api/post/single-post.php]: (Array) or {Array} I think we're doing?

Also: (Array) $data - *A*ll the GET data the is *sent* with the API request

File: api/post/single-post.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    // post_id variable from the get request
+    $post_id = $data['post_id'];
+
+    //gets all the data needed for the request
+    $sql = $wpdb->get_row($wpdb->prepare("SELECT $wpdb->posts.post_title as title, $wpdb->posts.post_date as date, $wpdb->users.display_name as author_name, $wpdb->posts.post_name as slug, $wpdb->posts.post_content as content FROM $wpdb->posts INNER JOIN $wpdb->users ON $wpdb->posts.post_author = $wpdb->users.ID WHERE $wpdb->posts.ID = %s", $post_id), ARRAY_A);
+
+    //assign all the variables to the array to be returned as a JSON response
+    $result['title'] = $sql['title'];
+    $result['content'] = $sql['content'];
+    $result['date'] = $sql['date'];
```
Comment (RollYourOwnEd) [inline api/post/single-post.php]: Format of this worth specifying

File: api/post/single-post.php (hunk 2 above)
Comment (richlloydmiles) [inline api/post/single-post.php]: Are you referring to the param not having surrounding brackets? Everywhere I can find it there is no brackets - it literally just has the parameter type - http://phpdoc2cheatsheet.com/, https://www.phpdoc.org/docs/latest/references/phpdoc/tags/param.html

This is not something that we have ever discussed from a coding standard perspective but probably worth the conversation.

File: api/post/single-post.php (hunk 3 above)
Comment (richlloydmiles) [inline api/post/single-post.php]: The date format?

File: api/post/single-post.php (hunk 1 above)
Comment (richlloydmiles) [inline api/post/single-post.php]: We haven't got that far yet. I assume this should be the case though.

**Comment by richlloydmiles** [comment]
Closing this in favor of https://github.com/Nona-Creative/go2africa/pull/5

**Pull request closed without merging**